fuel/app/classes/materia/widget/instance/manager.php

```diff
@@ -76,22 +77,23 @@ public static function get_all_for_user($user_id, $load_qset=false)
  *
  * @return array of widget instances that are visible to the user.
  */
-	public static function get_paginated_for_user($user_id, $page_number = 0)
+	public static function get_paginated_instances_for_user($user_id, $page_number = 0)
 	{
 		$inst_ids = Perm_Manager::get_all_objects_for_user($user_id, Perm::INSTANCE, [Perm::FULL, Perm::VISIBLE]);
-		$displayable_inst = self::get_all($inst_ids);
-		$widgets_per_page = 80;
-		$total_num_pages = ceil(sizeof($displayable_inst) / $widgets_per_page);
-		$offset = $widgets_per_page * $page_number;
-		$has_next_page = $offset + $widgets_per_page < sizeof($displayable_inst) ? true : false;
 
-		// inst_ids corresponds to a single page's worth of instances
-		$displayable_inst = array_slice($displayable_inst, $offset, $widgets_per_page);
+		$items_per_page = 80;
+		$offset = $items_per_page * $page_number;
+
+		// query DB for only a single page of instances + 1
+		$displayable_items = self::get_all($inst_ids, false, false, false, $offset, $items_per_page);
+
+		// if the returned number of instances is greater than a page, there's more pages
+		$has_next_page = sizeof($displayable_items) > ($items_per_page - 1) ? true : false;
```

Comment:
This is getting into the weeds a bit, but this method of determining whether there's another page will still return true if the total item count is divisible by `items_per_page`.

A better option might be to set the query limit to `$limit + 1`, and simply check whether `$displayable_items` is greater than `$items_per_page`. That shouldn't affect the offset, since the +1 isn't part of the calculation.

Reply:
Update: I'm also wondering if the behavior you noticed with `order by` was influenced by the +1 offset. I'd prefer to keep the `order by` clause when performing the query, so it'd be worth double-checking after updating these.

Reply:
Having a single `order_by` seems to have been causing the issue, since multiple instances can have the same `created_at` value. I added an additional criteria to break up ties by ordering by ID, and reinstated the original pagination logic (adding +1 to the limit and then popping the last entry).